### DISCUSSION TITLE
Android Studio Live Reload not working

### DISCUSSION THREAD

**Discussion by leo-petrucci** [ionic-team/capacitor#4123] [Q&A] [answered]
I'm testing out Capacitor 3.0 and I'm having trouble getting Android Studio live reload to work properly.

My `capacitor.config.ts` looks like this:

```
import { CapacitorConfig } from '@capacitor/cli';

const config: CapacitorConfig = {
  appId: 'uk.co.example',
  appName: 'example',
  webDir: 'build',
  bundledWebRuntime: false,
  server: {
    url: 'http://localhost:3000',
    cleartext: true,
  },
};

export default config;
```

I have run both `npx cap sync` as well as `npx cap copy`.

Live reload works perfectly in Xcode, but when launching my app on Android Studio it's running the production build rather than the one running at `http://localhost:3000`. Is this a bug or am I doing it wrong?

**Comment by imhoffd** [comment]
Is it writing the `server.url` to `android/app/src/main/assets/capacitor.config.json`?

**Comment by jcesarmobile** [comment, marked as answer]
Are you testing on real devices or simulators?
`http://localhost:3000` means "this computer on port 3000"

So, on iOS simulator it works because the simulator runs on the same machine as the server (your computer)
On Android emulator it won't work because it's behind a virtual [router/firewall](https://developer.android.com/studio/run/emulator-networking)
> Each instance of the emulator runs behind a virtual router/firewall service that isolates it from your development machine network interfaces and settings and from the internet. An emulated device can't see your development machine or other emulator instances on the network. Instead, it sees only that it is connected through Ethernet to a router/firewall.

On real devices it won't work because the server is running in the computer and "localhost" is "this device", so fails because you don't have a server running on the real device.

In those cases, you should use the local network IP (if the devices are on the same network), something like `192.168.1.XX`, and as the local IP works for all cases it's recommended to always use that instead of localhost.

**Reply by leo-petrucci** [reply]
Ah, I see. Unfortunately running it on `192.168.1.XX` breaks my cookies so I'm kind of at an empasse. Thank you for the response!
